### ISSUE TITLE
Can't get Shadows on Flat surfaces to reliably look good

### ISSUE THREAD

**Issue by Sparkky** [dustyroom-studio/quibli-doc#11]
**Describe the bug**
I can't seem to find any way to tweak the Shadows in the Quibli URP Config so that flat edged objects don't have this visual glitch at some angles.

**To Reproduce**
Setup a Scene with Quibli's default values, add a directional light and the default Unity Squares - apply a material using the Quibli / Stylized Lit shader.
Rotate the Light source (or the square object) and observe very reliable and obvious shadow artifacts 

**Expected behavior**
With the other shaders I had previously used (and ones I wrote myself that are similar but less feature rich than Quibli) I was able to tweak the depth & normal bias to remove these artifacts, with very minimal distortion - with quibli even if I max out the depth and normal bias (which looks awful) the artifacts are still very visible. Even if I create an unrealisting scenario with the cascades where the boxes in question should have far more resolution than they would need.

It appears the issue is only related to the self casted shadows, nothing else seems to suffer from these artifacts.

**Screenshots**
![image](https://user-images.githubusercontent.com/1709027/168910791-3c4a89dc-9764-4529-a0ed-0a9f31f430bc.png)

**Unity details**:
Quibli version 1.4.8
Unity 2021.3.0f1
Dev platform: WindowsEditor
Target platform: StandaloneWindows64
URP installed: True, version 12.1.6
Render pipeline: UniversalPipeline
Color space: Linear
Quality config: Quibli URP Config
Graphics config: Quibli URP Config

**Additional context**
For my game I have a slowly rotating directional light source over the course of the day - I was using the real time received shadow so that as objects are behind shadow (such as under a tree or next to a cliff/wall) appear darkened. If there is another way you would recommend to accomplish with your shaders, I am very open to a work around, I've just never had an issue with this with a shader and after several hours of researching I can't seem to come up with a good alternative.

Just incase it helps, this is another example where I'd want the block to be partially shadowed by the wall, but the self-shadow is showing the artifacts 
![image](https://user-images.githubusercontent.com/1709027/168912312-54ba96f4-8efe-4661-afad-0c8338e49c25.png)

**Comment by dustyroom-studio** [comment]
Hi,

Thank you very much for such a detailed report.

First of all, the problem you are asking about is a commonly known issue with Unity, in particular — with realtime shadows on the flat surfaces (as opposed to curved surfaces — see pic below). This 'bug' is just emphasized in the stylized shaders. 

<img width="407" alt="unity-urp-shadows-flat-curved-surfaces" src="https://user-images.githubusercontent.com/74666746/169007549-80040b7e-888e-4940-b791-ab51b75b25fe.png">

Unity's realtime shadows basically use a 'live' texture and you have no upscale options besides the Shadow Resolution parameter in the URP settings:

<img width="397" alt="unity-urp-settings-shadow-resolution" src="https://user-images.githubusercontent.com/74666746/169006040-f7f6f158-90cf-4a85-acdd-b85df6bb8528.png">

One of the workarounds I can think of for now is the following:
[Video](https://drive.google.com/file/d/123N4R3X0m_T1hWCP2xGPH2T206bKymmh/view?usp=sharing)

Basically,
- Disable the Cast Shadows parameter on the cube
- Duplicate it, slightly reduce it and nest it inside the big shadowless cube
- Turn on the Cast Shadows on the smaller nested cube

Thank you for bringing this to our attention. Although this is a quirk of the Unity's realtime shadows, we'll see if there is anything we can do about it within Quibli.
